src/GraphQL/Instrumentation/TelemetryProvider.cs

```diff
@@ -0,0 +1,135 @@
+#if NET5_0_OR_GREATER
+
+using System.Diagnostics;
+using GraphQL.DI;
+using GraphQL.Execution;
+using GraphQL.Types;
+using GraphQL.Validation;
+using GraphQLParser.AST;
+
+namespace GraphQL.Instrumentation;
+
+/// <summary>
+/// Provides telemetry through the <see cref="Activity">System.Diagnostics.Activity API</see> as an <see cref="IConfigureExecution"/> implementation.
+/// Derive from this class to add additional telemetry.
+/// <para>
+/// To use a derived class, call <see cref="GraphQLBuilderExtensions.ConfigureExecution{TConfigureExecution}(IGraphQLBuilder)"/> with the type of your derived class.
+/// </para>
+/// </summary>
+public class TelemetryProvider : IConfigureExecution
+{
+    private readonly ActivitySource _activitySource;
+    private readonly TelemetryOptions _telemetryOptions;
+    private const string ACTIVITY_OPERATION_NAME = "graphql";
+
+    /// <summary>
+    /// Gets the source name used for telemetry.
+    /// </summary>
+    public static string SourceName => "GraphQL";
+
+    /// <summary>
+    /// Initializes a new instance with the specified options.
+    /// </summary>
+    public TelemetryProvider(TelemetryOptions options)
+    {
+        _activitySource = new ActivitySource(SourceName, typeof(TelemetryProvider).Assembly.GetNuGetVersion());
```

Comment:
Let's move this to field initializer and make `_activitySource` field static.

Reply:
Done